lib/matplotlib/cbook/__init__.py

```diff
@@ -236,18 +162,16 @@ def __setstate__(self, state):
     def connect(self, s, func):
         """Register *func* to be called when signal *s* is generated.
         """
-        self._func_cid_map.setdefault(s, WeakKeyDictionary())
-        # Note proxy not needed in python 3.
-        # TODO rewrite this when support for python2.x gets dropped.
-        proxy = _BoundMethodProxy(func)
+        self._func_cid_map.setdefault(s, {})
```

Comment:
pros / cons of using `ismthod` vs `try:....except:...` on attempting to create the `WeakMethod`?

Reply:
Edited accordingly.
As mentioned elsewhere I believe quite strongly that the use of weakmethod semantics is pretty silly anyways...